streams/client/client.go

```diff
@@ -112,56 +122,76 @@ func min(t0 time.Time, t1 time.Time) time.Time {
 }
 
 func (c *Client) Send(cmdName string, body command.Body) (cr.Response, error) {
-	_, response, err := c.send(cmdName, body)
-	return response, err
-}
+	responses, err := c.send(cmdName, body, 1)
+	if err != nil {
+		return nil, err
+	}
 
-func (c *Client) SendForPid(cmdName string, body command.Body) (peerCore.ID, cr.Response, error) {
-	return c.send(cmdName, body)
+	resp, ok := <-responses
+	if !ok {
+		return nil, errors.New("timeout")
+	}
+
+	if resp.error != nil {
+		return nil, resp.error
+	}
+
+	return resp.Response, nil
 }
 
-func (c *Client) send(cmdName string, body command.Body) (peerCore.ID, cr.Response, error) {
-	cmd := command.New(cmdName, body)
-	now := time.Now()
-	ctx, ctx_close := context.WithDeadline(c.ctx, now.Add(SendToPeerTimeout))
-	defer ctx_close()
-	strmDD, _ := ctx.Deadline()
+func (c *Client) MultiSend(cmdName string, body command.Body, thresh int) (map[peerCore.ID]cr.Response, map[peerCore.ID]error, error) {
+	responses, err := c.send(cmdName, body, thresh)
+	if err != nil {
+		return nil, nil, err
+	}
+	rets := make(map[peerCore.ID]cr.Response)
+	errs := make(map[peerCore.ID]error)
+	for resp := range responses {
+		if resp.error != nil {
+			errs[resp.ID] = resp.error
+		} else {
```

Comment:
Is there instance where there is error and response?

Reply:
yup